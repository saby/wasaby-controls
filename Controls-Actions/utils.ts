import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
