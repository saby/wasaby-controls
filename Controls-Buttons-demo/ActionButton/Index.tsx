import ButtonType from 'Controls-Buttons-meta/buttonConnectedButtonType';
import { Button } from 'Controls-Buttons/buttonConnected';
import { PropsDemoEditor } from 'Controls-demo/ObjectTypeEditor/PropsDemoEditor';
import { Provider as DataContextProvider } from 'Controls-DataEnv/context';
import { RecordSet } from 'Types/collection';
import { actions } from './actions';

function ButtonPropsEditor() {
    const contextConfig = {
        Actions: {
            dataFactoryName: 'Controls/dataFactory:List',
            dataFactoryArguments: {
                keyProperty: 'id',
                displayProperty: 'commandName',
                groupProperty: 'category',
            },
        },
    };

    const contextData = {
        Actions: {
            items: new RecordSet({
                keyProperty: 'id',
                rawData: actions,
            }),
        },
    };
    return (
        <DataContextProvider configs={contextConfig} loadResults={contextData}>
            <PropsDemoEditor metaType={ButtonType} control={Button} />;
        </DataContextProvider>
    );
}

export default ButtonPropsEditor;
