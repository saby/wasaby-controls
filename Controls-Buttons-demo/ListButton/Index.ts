import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import controlTemplate = require('wml!Controls-Buttons-demo/ListButton/Template');
import { RecordSet } from 'Types/collection';
import 'wml!Controls-Buttons-demo/ListButton/resources/template';

export default class ListButton extends Control<IControlOptions> {
    protected _template: TemplateFunction = controlTemplate;
    protected _items: RecordSet = null;
    protected _menuOptions: Object;

    protected _beforeMount(): void {
        this._items = new RecordSet({
            keyProperty: 'id',
            rawData: [
                {
                    id: 0,
                    title: 'Калорийность',
                    additionalText: 'Белки, жиры, углеводы',
                    template: 'wml!Controls-Buttons-demo/ListButton/resources/template',
                },
                {
                    id: 1,
                    title: 'Размеры',
                    additionalText: 'Длинна, ширина, высота',
                    template: 'wml!Controls-Buttons-demo/ListButton/resources/template',
                },
                {
                    id: 2,
                    title: 'Производитель',
                    additionalText: 'Страна, адрес, реквизиты',
                    template: 'wml!Controls-Buttons-demo/ListButton/resources/template',
                },
            ],
        });

        this._menuOptions = {
            items: this._items,
            itemTemplateProperty: 'template',
            keyProperty: 'id',
        };
    }
}
