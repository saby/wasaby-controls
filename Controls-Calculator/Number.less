.calculator-View__number__clickable {
   cursor: pointer;
}
