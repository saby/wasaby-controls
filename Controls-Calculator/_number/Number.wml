<ws:partial scope="{{ _options }}"
            template="{{_options.content}}"
            on:mousedown="_clickHandler()"
            on:click="_clickHandler()"
            attr:class="controls_Calculator_theme-{{_options.theme}} calculator-View__number
            {{_isCalculatorRegistered ? 'calculator-View__number__clickable'}}"
/>
