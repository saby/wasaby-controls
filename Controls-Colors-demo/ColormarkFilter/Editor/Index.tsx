import * as React from 'react';
import { Memory } from 'Types/source';
import { IDataConfig, IListDataFactoryArguments } from 'Controls-DataEnv/dataFactory';
import { View as FilterView } from 'Controls-ListEnv/filterConnected';

const filterItems = [
    {
        id: '4',
        icon: 'Flag',
        iconStyle: 'primary',
        iconSize: 'm',
        caption: 'Важный',
    },
    {
        id: '5',
        icon: 'Add',
        iconStyle: 'primary',
        iconSize: 'm',
        caption: 'Плюс',
    },
    {
        id: '0',
        type: 'color',
        value: {
            color: '--danger_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Красный',
        removable: true,
        editable: true,
    },
    {
        id: '1',
        type: 'color',
        value: {
            color: '--success_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Зеленый',
    },
    {
        id: '2',
        type: 'color',
        value: {
            color: '--secondary_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Синий',
        removable: true,
        editable: true,
    },
    {
        id: '3',
        type: 'color',
        value: {
            color: '--warning_color',
            style: {
                b: true,
                i: true,
                u: true,
                s: true,
            },
        },
        caption: 'Желтый',
        removable: true,
        editable: true,
    },
];

export default class extends React.Component {
    static getLoadConfig(): Record<string, IDataConfig<IListDataFactoryArguments>> {
        return {
            markData: {
                dataFactoryName: 'Controls/dataFactory:List',
                dataFactoryArguments: {
                    source: new Memory({
                        data: [],
                        keyProperty: 'id',
                    }),
                    keyProperty: 'id',
                    displayProperty: 'title',
                    filterDescription: [
                        {
                            name: 'mark',
                            editorTemplateName: 'Controls-Colors/colormarkFilter:Editor',
                            resetValue: { selected: [], excluded: [] },
                            viewMode: 'extended',
                            value: { selected: ['1', '3'], excluded: ['0', '2'] },
                            editorOptions: {
                                items: filterItems,
                                extendedCaption: 'Пометки',
                            },
                        },
                    ],
                },
            },
        };
    }

    render() {
        return (
            <div>
                <FilterView ref={this.props.forwardedRef} storeId="markData" />
            </div>
        );
    }
}
