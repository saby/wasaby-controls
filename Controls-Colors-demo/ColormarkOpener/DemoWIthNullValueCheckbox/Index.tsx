import { forwardRef, LegacyRef, ReactElement, useState, useCallback } from 'react';
import { useTheme } from 'UICore/Contexts';
import { itemsWithPartialSelected, palette } from '../../data';
import { DialogOpener } from 'Controls-Colors/colormarkOpener';
import { ItemsView } from 'Controls/list';
import { list } from '../resources/data';
import { Model } from 'Types/entity';
import ItemTemplate from '../templates/ItemTemplate';
import 'css!Controls-Colors-demo/Style';

const helper = new DialogOpener({});
const itemActions = [
    {
        id: 1,
        tooltip: 'Выбрать пометку',
        icon: 'icon-Colorize',
    },
];

export default forwardRef((_: unknown, ref: LegacyRef<HTMLDivElement>): ReactElement => {
    const theme = useTheme();

    const [listMarkedKey, setListMarkedKey] = useState('1');

    const onMarkedKeyChangeHandler = useCallback((key) => {
        setListMarkedKey(key);
    }, []);

    const actionClickHandler = useCallback(
        (_: Event, item: Model<{ keys: string[] }>) => {
            helper.open({
                templateOptions: {
                    items: itemsWithPartialSelected,
                    palette,
                    selectedKeys: item.get('keys') || [],
                    isAdaptive: true,
                    multiSelect: true,
                    onSelectedKeysChanged(keys) {
                        itemsWithPartialSelected.forEach((item) => {
                            if (!keys.includes(item.id)) {
                                item.partialSelected = undefined;
                            }
                        });
                    },
                },
                width: theme === 'default' ? 252 : 405,
                eventHandlers: {
                    onResult(res: string[]) {
                        item.set('keys', res);
                    },
                },
            });
        },
        [theme]
    );

    return (
        <div className="tw-flex tw-justify-center" ref={ref}>
            <div
                className="tw-flex tw-justify-baseline controls-cursor_pointer Controls-Colors-demo_border
                                Controls-Colors-demo_list"
            >
                <ItemsView
                    items={list}
                    className="tw-w-full"
                    itemActions={itemActions}
                    markedKey={listMarkedKey}
                    onActionClick={actionClickHandler}
                    onMarkedKeyChanged={onMarkedKeyChangeHandler}
                    itemTemplate={ItemTemplate}
                    itemTemplateOptions={{ items: itemsWithPartialSelected }}
                />
            </div>
        </div>
    );
});
