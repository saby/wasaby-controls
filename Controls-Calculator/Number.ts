import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!Controls-Calculator/_number/Number';
import { Bus } from 'Env/Event';
import { SyntheticEvent } from 'UICommon/Events';
import { Guid } from 'Types/entity';
import { Opener as CalculatorOpener } from 'Controls-Calculator/Button';

/**
 * Контрол для отображения числа, вставляемого в калькулятор.
 * @control
 * @extends UI/Base:Control
 * @demo Controls-Calculator-demo/ClickOnNumber/ClickOnNumber
 * @public
 */
export default class Number extends Control {
    protected _template: TemplateFunction = template;
    protected _isCalculatorRegistered: boolean = false;
    protected _id: string;

    protected _afterMount(): void {
        this._id = Guid.create();
        this._notify('CalculatorRegisterNumber', [this, this._id], {
            bubbling: true,
        });
    }

    protected _clickHandler(event: SyntheticEvent): void {
        if (this._needProcessClick()) {
            event.stopPropagation();
            Bus.channel('CalculatorEvents').notify('numberClick', this._container.textContent);
        }
    }

    protected _beforeUnmount(): void {
        this._notify('CalculatorUnregisterNumber', [this._id], {
            bubbling: true,
        });
    }

    private _needProcessClick(): boolean {
        // Калькулятор может быть не в окне, только опенер проверять недостаточно.
        return this._isCalculatorRegistered || CalculatorOpener.isOpened();
    }

    setCalculatorRegistered(state: boolean): void {
        this._isCalculatorRegistered = state;
    }

    static _styles: string[] = ['Controls-Calculator/Number'];
}
