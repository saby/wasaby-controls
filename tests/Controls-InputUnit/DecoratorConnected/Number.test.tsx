/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';
import { getContent } from '../resources/utils';
import { default as Index } from 'Controls-Input-demo/DecoratorConnected/Number/Index';

describe(
    'Controls-Input/decoratorConnected:Number',
    () => {
        let container = null;

        beforeEach(() => {
            jest.useFakeTimers('modern');
            jest.setSystemTime(new Date(1996, 5, 21));
            container = document.createElement('div');
            WasabyEvents.initInstance(container);
            document.body.appendChild(container);
        });
        afterEach(() => {
            unmountComponentAtNode(container);
            WasabyEvents.destroyInstance(container);
            jest.useRealTimers();
            container.remove();
            container = null;
        });

        it('Index', async () => {
            render(getContent(Index), {container});
            expect(container).toMatchSnapshot();
        });
    }
);
