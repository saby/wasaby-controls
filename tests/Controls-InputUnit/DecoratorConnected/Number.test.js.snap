// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/decoratorConnected:Number Index 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <span
      class="controls-DecoratorNumber controls-fontweight-default"
    >
      22
    </span>
  </div>
</div>
`;
